org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinWhenFilter.java

```diff
@@ -11,18 +11,26 @@
  *******************************************************************************/
 package org.jacoco.core.internal.analysis.filter;
 
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.tree.AbstractInsnNode;
 import org.objectweb.asm.tree.InsnNode;
 import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.LookupSwitchInsnNode;
 import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.tree.TableSwitchInsnNode;
 
 /**
  * Filters bytecode that Kotlin compiler generates for <code>when</code>
- * expressions which list all cases of <code>sealed class</code>, i.e. which
- * don't require explicit <code>else</code>.
+ * expressions which list all cases of <code>enum</code> or
+ * <code>sealed class</code>, i.e. which don't require explicit
+ * <code>else</code>.
  */
-public final class KotlinWhenSealedFilter implements IFilter {
+public final class KotlinWhenFilter implements IFilter {
```

Comment:
@marchof we'll have `KotlinWhenStringFilter` and `KotlinWhenFilter`, but I didn't managed to find better name